File: leet125/leet125.py
# https://leetcode.com/problems/valid-palindrome/

class Solution:
    def isPalindrome(self, s: str) -> bool:
        clean_s = ""
        for i in s:
            if i.isalnum():
                clean_s += i.lower()
        left, right = 0, len(clean_s) - 1
        while left < right:
            if clean_s[left] != clean_s[right]:
                return False
            left += 1
            right -= 1
        return True

File: leet125/leet125.cpp
// https://leetcode.com/problems/valid-palindrome/description/
#include <string>
using namespace std;

class Solution
{
public:
    bool isPalindrome(string s)
    {
        string clean_s = "";
        for (int i = 0; i < s.length(); i++)
        {
            if (isalnum(s[i]))
            {
                clean_s += tolower(s[i]);
            }
        }

        int left = 0;
        int right = clean_s.length() - 1;

        while (left < right)
        {
            if (clean_s[left] != clean_s[right])
            {
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
};
